sync-mixins.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TMPDIR="./tmp"
+MIXIN_REPO="git@github.com:giantswarm/giantswarm-kubernetes-mixin.git"
+# clone a branch if provided
+BRANCH=${1:-""}
+
+# make a temporary dir to work in
+mkdir ${TMPDIR}
+
+if [ -z "${BRANCH}" ]; then
+    # clone the mixins repo
+    echo -e "\nCloning master branch:\n"
+    git clone ${MIXIN_REPO} ${TMPDIR}/mixins
```

Comment:
I think you can also use `--single-branch` here

```suggestion
    git clone --single-branch ${MIXIN_REPO} ${TMPDIR}/mixins
```